### DISCUSSION TITLE
[resemblejs] needs fix

### DISCUSSION THREAD

**Discussion by GreenYun** [DefinitelyTyped/DefinitelyTyped#59325] [Issues with a @types package]
Please fix type declarations of `resemblejs` according to its source code.

I have already written some codes about `resemblejs/compareImages` (yes, it is a separate module now), and improved the type binding of `ResembleSingleCallbackComparisonOptions.ignore`. It should accept a string array. My fix as the following:
```typescript
    type ResembleComparisonCallbackIgnoreOption = "nothing" | "less" | "antialiasing" | "colors" | "alpha";

    interface ComparisonCallbackOptions {
      output?: OutputSettings | undefined;
      returnEarlyThreshold?: number | undefined;
      scaleToSameSize?: boolean | undefined;
      ignore?: ResembleComparisonCallbackIgnoreOption | Array<ResembleComparisonCallbackIgnoreOption> | undefined;
    }
```

Also, please consider simplifying the identifiers of interfaces. Due to namespace defined, I think we should not add type prefixes.

[a comment by an automated account is not shown]

**Comment by pimterry** [comment]
I'm the only owner of these types, and I haven't used Resemble in many years, so I'm not going to fix this I'm afraid.

If you're interested in fixing the resemblejs types, your best option is to open a PR with a patch to make these changes. There's more info on how to do that in the README: https://github.com/DefinitelyTyped/DefinitelyTyped#how-can-i-contribute. You're welcome to change the module owner along the way, I'm not in a position to maintain this myself any more.
